src/models/appointment.ts

```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+
+export interface IAppointment {
+    patient: string;
+    doctor: string;
+    date: Date;
+    time: string;
+    type: string;
+    link: string;
+}
+
+export interface IAppointmentModel extends IAppointment, mongoose.Document { }
+
+const appointmentSchema = new mongoose.Schema({
+    patient: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true,
+    },
+    doctor: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true,
+    },
+    date: {
+        type: Date,
+        required: true,
+    },
+    time: {
+        type: String,
+        required: true,
+    },
+    type: {
+        type: String,
+        required: true,
+        enum: ["online", "face2face"],
+        default: "online",
+    },
+    link: {
+        type: String,
```

Comment:
what is link column used for?

Reply:
I assume we will use 3rd party applications for meetings ( such as Zoom, Google Meet, Microsoft Teams ). If so, we can share the invitation to the user using this string. For better understanding, we can use name invitation instead of link too.

Reply:
But i think patient or doctor should be able to see their incoming appointments in the app. Since link is not required, we will add it if it is necessary. It will not be extra column in data. It will be an extra attribute for model.

Reply:
Ok, that means that in the future there will be a separated input for the doctors to add a link to the appointment if it is online, right?

Reply:
Yes, nurse doctor or admin can add link if it is needed. Or, if patient wants to meet online with doctor and select available time from doctor's calendar, we can set a meeting link and store it here.